### PR TITLE
Fix cfg dominator top

### PR DESCRIPTION
**Pull request by martin-cs**
There are a couple of linting failures but these are due to a slightly over-zealous standard IMHO.

@danpoe This is one of the features I added at Toyota and we need for the next release.

### CODE REVIEW

File: src/analyses/cfg_dominators.h [hunk 1]
```diff
@@ -6,8 +6,8 @@ Author: Georg Weissenbacher, georg@weissenbacher.name
 
 \*******************************************************************/
 
-#ifndef CPROVER_ANALYSES_CFG_DOMINATORS_H
-#define CPROVER_ANALYSES_CFG_DOMINATORS_H
+#ifndef CPROVER_SRC_ANALYSES_CFG_DOMINATORS_H
+#define CPROVER_SRC_ANALYSES_CFG_DOMINATORS_H
```
Comment (tautschnig) [inline src/analyses/cfg_dominators.h]: This seems wrong, does the hook need updating?

File: src/analyses/cfg_dominators.h [hunk 2]
```diff
@@ -170,10 +167,14 @@ void cfg_dominators_templatet<P, T, post_dom>::fixedpoint(P &program)
...
       while(n_it!=node.dominators.end() && o_it!=other.end())
       {
-        if(*n_it==current) ++n_it;
-        else if(*n_it<*o_it) { changed=true; node.dominators.erase(n_it++); }
-        else if(*o_it<*n_it) ++o_it;
-        else { ++n_it; ++o_it; }
+        if(*n_it==current)
+          ++n_it;
+        else if(*n_it<*o_it)
+          { changed=true; node.dominators.erase(n_it++); }
```
Comment (tautschnig) [inline src/analyses/cfg_dominators.h]: { in the same column as else, newlines

File: src/analyses/cfg_dominators.h [hunk 3]
```diff
@@ -35,7 +35,6 @@ class cfg_dominators_templatet
 
   void operator()(P &program);
 
-  target_sett top;
```
Comment (tautschnig) [inline src/analyses/cfg_dominators.h]: Should the commit message be "Remove unused top" ?

**Comment by martin-cs** [comment]
Rebased which fixes the compilation and linting issues.  Patch is now near trivial but has significant performance impact on some code.

**Comment by peterschrammel** [comment]
@kroening, @tautschnig, I think this can be merged.

**Pull request merged**